lib/internal/util.js

```diff
@@ -850,10 +851,37 @@ const encodingsMap = { __proto__: null };
 for (let i = 0; i < encodings.length; ++i)
   encodingsMap[encodings[i]] = i;
 
+/**
+ * Reassigns the .name property of a function.
+ * Should be used when function can't be initially defined with desired name
+ * or when desired name should include `#`, `[`, `]`, etc.
+ * @param {string | symbol} name
+ * @param {Function} fn
+ * @param {object} [descriptor]
+ * @returns {Function} the same function, renamed
+ */
+function assignFunctionName(name, fn, descriptor = kEmptyObject) {
+  if (typeof name === 'symbol') {
```

Comment:
nit: we could also ensure the code throws if `name` is not a symbol nor a string by changing the condition here
```suggestion
  if (typeof name !== 'string') {
```

Reply:
This would make it the potential error look strange (something like `TypeError: Symbol.prototype.description requires that 'this' be a Symbol`).
Since this is internal function that is not exposed outside, I'd rather keep the `if` condition unambiguous and assume the type is always correct.

Reply:
I don't understand the reasoning, given that it's an internal thing I would say that we definitely want it to throw so we're more likely to spot it before it gets in a release, but I don't feel strongly

Reply:
On a second thought, I agree with this change. Given the `string | symbol` type, it's unambiguous.
The reasoning was based on assumption that it would need an additional assertion, i.e.
```js
if (typeof name === 'symbol') {
  // logic for symbol
}
```
Would change to
```js
if (typeof name !== 'string') {
  // logic for any non-string
  assert(typeof name === 'symbol'); // explicit assertion
  // logic for symbol
}
```
Which is not really necessary.